Test runner for k8 fails if version changes

steps to reproduce it:
1. bump up `VERSION` file to `0.7.0-alpha.6`
2. `make smoke-test-k8`

This fails:
```
Sending build context to Docker daemon  208.9MB
Step 1/2 : FROM alpine:3.12
 ---> 389fef711851
Step 2/2 : COPY fluvio fluvio
 ---> 020bca8d1b89
Successfully built 020bca8d1b89
Successfully tagged infinyon/fluvio:e577864e815b4b1eb8150876cf73a2fbd8500981
FLUVIO_CMD=true ./target/debug/flv-test --spu 1 --produce-iteration 1000 --develop --client-log warn --server-log debug --skip-checks
Start running fluvio test runner
using k8 environment driver
installing cluster
panic panicked at 'called `Result::unwrap()` on an `Err` value: InstallK8(HelmChartNotFound("fluvio/fluvio-app:0.7.0-alpha.6"))', tests/runner/src/environment/k8.rs:47:42
Makefile:78: recipe for target 'smoke-test-k8' failed
```

--- a/src/cluster/src/cli/start/k8.rs
+++ b/src/cluster/src/cli/start/k8.rs
@@ -1,8 +1,4 @@
-use std::io::Error as IoError;
-use std::io::ErrorKind;
 use std::convert::TryInto;
-use std::process::Command;
-
 use fluvio::config::TlsPolicy;
 
 use crate::{ClusterInstaller, ClusterError, K8InstallError, StartStatus, ClusterConfig};
@@ -27,6 +23,11 @@ pub async fn process_k8(
         .unwrap_or(default_chart_version);
 
     let mut builder = ClusterConfig::builder(chart_version);
+
+    if opt.develop {
+        builder.development()?;
+    }
+
     builder
         .namespace(opt.k8_config.namespace)
         .group_name(opt.k8_config.group_name)
@@ -39,48 +40,24 @@ pub async fn process_k8(
         .with_if(skip_sys, |b| b.install_sys(false))
         .with_if(opt.skip_checks, |b| b.skip_checks(true));
 
-    match opt.k8_config.chart_location {
-        // If a chart location is given, use it
-        Some(chart_location) => {
-            builder.local_chart(chart_location);
-        }
-        // If we're in develop mode (but no explicit chart location), use hardcoded local path
-        None if opt.develop => {
-            builder.local_chart("./k8-util/helm");
-        }
-        _ => (),
+    if let Some(chart_location) = opt.k8_config.chart_location {
+        builder.local_chart(chart_location);
     }
 
-    match opt.k8_config.registry {
-        // If a registry is given, use it
-        Some(registry) => builder.image_registry(registry),
-        None => builder.image_registry("infinyon"),
-    };
+    if let Some(registry) = opt.k8_config.registry {
+        builder.image_registry(registry);
+    }
 
     if let Some(rust_log) = opt.rust_log {
         builder.rust_log(rust_log);
     }
+
     if let Some(map) = opt.authorization_config_map {
         builder.authorization_config_map(map);
     }
 
-    match opt.k8_config.image_version {
-        // If an image tag is given, use it
-        Some(image_tag) => {
-            builder.image_tag(image_tag.trim());
-        }
-        // If we're in develop mode (but no explicit tag), use current git hash
-        None if opt.develop => {
-            let output = Command::new("git").args(&["rev-parse", "HEAD"]).output()?;
-            let git_hash = String::from_utf8(output.stdout).map_err(|e| {
-                IoError::new(
-                    ErrorKind::InvalidData,
-                    format!("failed to get git hash: {}", e),
-                )
-            })?;
-            builder.image_tag(git_hash.trim());
-        }
-        _ => (),
+    if let Some(image_tag) = opt.k8_config.image_version {
+        builder.image_tag(image_tag.trim());
     }
 
     let config = builder.build()?;

--- a/src/cluster/src/start/k8.rs
+++ b/src/cluster/src/start/k8.rs
@@ -38,6 +38,8 @@ const DEFAULT_CHART_APP_NAME: &str = "fluvio/fluvio-app";
 const DEFAULT_GROUP_NAME: &str = "main";
 const DEFAULT_CLOUD_NAME: &str = "minikube";
 const DEFAULT_SPU_REPLICAS: u16 = 1;
+const DEFAULT_DEV_CHART: &str = "./k8-util/helm";
+
 const FLUVIO_SC_SERVICE: &str = "fluvio-sc-public";
 /// maximum time waiting for sc service to come up
 static MAX_SC_SERVICE_WAIT: Lazy<u64> = Lazy::new(|| {
@@ -388,6 +390,30 @@ impl ClusterConfigBuilder {
             .build_impl()
             .map_err(K8InstallError::MissingRequiredConfig)?;
         Ok(config)
+    }
+
+    /// Applies development options to this cluster configuration.
+    ///
+    /// The following options are applied when using this method.
+    /// NOTE that these options will be overwritten even if they were
+    /// previously assigned.
+    ///
+    /// - Use local_chart "./k8-util/helm"
+    /// - Use the git hash of HEAD as the image_tag
+    pub fn development(&mut self) -> Result<&mut Self, ClusterError> {
+        let output = Command::new("git")
+            .args(&["rev-parse", "HEAD"])
+            .result()
+            .map_err(K8InstallError::CommandError)?;
+        let git_hash = String::from_utf8(output.stdout).map_err(|e| {
+            K8InstallError::IoError(IoError::new(
+                ErrorKind::InvalidData,
+                format!("failed to get git hash: {}", e),
+            ))
+        })?;
+        self.image_tag(git_hash.trim());
+        self.local_chart(DEFAULT_DEV_CHART);
+        Ok(self)
     }
 
     /// Sets a local helm chart location to search for Fluvio charts.

--- a/src/sc/build.rs
+++ b/src/sc/build.rs
@@ -1,4 +1,5 @@
 fn main() {
     // Copy VERSION file. Do not fail e.g. when built via `cargo publish`
+    println!("cargo:rerun-if-changed=../../VERSION");
     let _ = std::fs::copy("../../VERSION", "./src/VERSION");
 }

--- a/tests/runner/src/environment/k8.rs
+++ b/tests/runner/src/environment/k8.rs
@@ -24,6 +24,9 @@ impl EnvironmentDriver for K8EnvironmentDriver {
 
     async fn start_cluster(&self) {
         let mut builder = ClusterConfig::builder(crate::VERSION);
+        if self.option.develop_mode() {
+            builder.development().expect("should test in develop mode");
+        }
         builder
             .spu_replicas(self.option.spu)
             .skip_checks(self.option.skip_checks())

